flake.nix

```diff
@@ -0,0 +1,21 @@
+{
+  description = "FormalSystems lean implementation";
+
+  inputs.lean.url = "github:leanprover/lean4";
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+
+  outputs = { self, lean, flake-utils }: flake-utils.lib.eachDefaultSystem (system:
+    let
+      leanPkgs = lean.packages.${system};
+      pkg = leanPkgs.buildLeanPackage {
+        name = "FormalSystems";
+        src = ./.;
+      };
+    in {
+      packages = pkg // {
+        inherit (leanPkgs) lean;
+      };
+
+      defaultPackage = pkg.modRoot;
```

Comment:
`defaultPackage` is deprecated in favour of `packages.default`, cf. NixOS/nix#5532